src/query_graph/graph_path.rs

```diff
@@ -189,6 +193,22 @@ pub(crate) struct ClosedBranch(pub(crate) Vec<Arc<ClosedPath>>);
 /// partial/open path in a GraphQL operation (i.e. one that does not end in a leaf field).
 pub(crate) struct OpenBranch(Vec<SimultaneousPathsWithLazyIndirectPaths>);
 
+impl<TTrigger, TEdge> GraphPath<TTrigger, TEdge>
+where
+    TTrigger: Eq + Hash,
+    TEdge: Copy + Into<Option<EdgeIndex>>,
+{
+    pub(crate) fn iter(&self) -> impl Iterator<Item = GraphPathItem<'_, TTrigger, TEdge>> {
+        // FIXME: are these 3 `Vec`s expected to always have the same length?
+        self.edges
+            .iter()
+            .copied()
+            .zip(&self.edge_triggers)
+            .zip(&self.edge_conditions)
+            .map(|((edge, trigger), condition)| (edge, trigger, condition))
```

Comment:
`zip` stops when either input iterator stops and silently truncates the rest, so it’s best with inputs of equal lengths. Is this the case here?

Reply:
Yeah, those three `Vec`s should always be the same length.